x-pack/plugins/lens/public/xy_visualization_plugin/xy_visualization.tsx

```diff
@@ -7,69 +7,42 @@
 import React from 'react';
 import { render } from 'react-dom';
 import { Position } from '@elastic/charts';
-import { Visualization, Operation } from '../types';
+import uuid from 'uuid';
 import { getSuggestions } from './xy_suggestions';
-import { XYArgs } from './xy_expression';
-import { NativeRenderer } from '../native_renderer';
-
-export type State = XYArgs;
-export type PersistableState = XYArgs;
+import { XYConfigPanel } from './xy_config_panel';
+import { Visualization } from '../types';
+import { State, PersistableState } from './types';
 
 export const xyVisualization: Visualization<State, PersistableState> = {
   getSuggestions,
 
   initialize(state) {
     return (
       state || {
-        title: 'Empty line chart',
-        legend: { isVisible: true, position: Position.Right },
         seriesType: 'line',
-        splitSeriesAccessors: [],
-        stackAccessors: [],
+        title: 'Empty XY Chart',
+        legend: { isVisible: true, position: Position.Right },
         x: {
-          accessor: '',
+          accessor: uuid.v4(),
           position: Position.Bottom,
           showGridlines: false,
-          title: 'Uknown',
+          title: 'X',
         },
         y: {
-          accessors: [],
+          accessors: [uuid.v4()],
           position: Position.Left,
           showGridlines: false,
-          title: 'Uknown',
+          title: 'Y',
         },
+        splitSeriesAccessors: [],
+        stackAccessors: [],
       }
     );
   },
 
-  getPersistableState(state) {
-    return state;
-  },
+  getPersistableState: state => state,
 
-  renderConfigPanel: (domElement, props) => {
-    render(
-      <div>
-        XY Visualization
-        <NativeRenderer
-          nativeProps={{
-            columnId: 'col1',
-            filterOperations: (op: Operation) => true,
-            suggestedOrder: 1,
-          }}
-          render={props.datasource.renderDimensionPanel}
-        />
-        <NativeRenderer
-          nativeProps={{
-            columnId: 'col2',
-            filterOperations: (op: Operation) => true,
-            suggestedOrder: 2,
-          }}
-          render={props.datasource.renderDimensionPanel}
-        />
-      </div>,
-      domElement
-    );
-  },
+  renderConfigPanel: (domElement, props) => render(<XYConfigPanel {...props} />, domElement),
 
-  toExpression: state => '',
+  toExpression: () => '',
```

Comment:
Is generating a real expression here your next step? I know you already merged your expression code.

Reply:
Yeah. The expression will come along in a future PR.